astropy/coordinates/funcs.py

```diff
@@ -279,3 +254,120 @@ def get_constellation(coord, short_name=False, constellation_list='iau'):
         return names[0]
     else:
         return names
+
+
+def _concatenate_components(reps_difs, names):
+    """ Helper function for the concatenate function below. Gets and
+    concatenates all of the individual components for an iterable of
+    representations or differentials.
+    """
+    values = []
+    for name in names:
+        data_vals = []
+        for x in reps_difs:
+            data_val = getattr(x, name)
+            data_vals.append(data_val.reshape(1, ) if x.isscalar else data_val)
+        concat_vals = np.concatenate(data_vals)
+
+        # Hack because np.concatenate doesn't fully work with Quantity
+        if isinstance(concat_vals, u.Quantity):
+            concat_vals._unit = data_val.unit
+
+        values.append(concat_vals)
+
+    return values
+
+def concatenate_representations(reps):
```

Comment:
This needs to be in `__all__` to be in the public API, right?

Reply:
done